pkg/log/logshipper/logshipper.go

```diff
@@ -117,6 +118,10 @@ func (ls *LogShipper) Log(keyvals ...interface{}) error {
 	return ls.shippingLogger.Log(keyvals...)
 }
 
+func (ls *LogShipper) Writer() io.Writer {
```

Comment:
I assume this is exposed for `slog` -- why expose the Writer, and not an `slog.Handler`? (honest question -- I don't know why one over the other)

Reply:
I guess one reason to expose a Writer, and not Handler, is that changing the level is very tied to the handler. I think I'm going to drop a pile of comments about that elsewhere in this review

Reply:
yes, it is tied to the handler

Reply:
I think this should expose a Handler. Probably just by pulling in `slog.NewJSONHandler(logShipper.Writer(), &slog.HandlerOptions{...})`. 

And then we can also expose `shipper.SetSloggerLevel(...)` with something akin to:
```
        loggerLevel := new(slog.LevelVar)
        loggerLevel.Set(slog.LevelInfo)
        logger := slog.New(slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{Level: loggerLevel })
```

(yoinked from some personal project code)

Reply:
nice, I didn't realize you could do that, I was trying to reassign the level of the handler and it wasn't working so I assumed you had to replace the handler, this is way better! thank you

Reply:
updated this, exposed as handler and added version as log attribute